packages/expo-device/src/Device.ts

```diff
@@ -0,0 +1,116 @@
+}
+export const osBuildFingerprint = fingerprint;
+export const modelName = model;
+export const modelId = platformModelId;
+export const designName = platformDesignName;
+export const productName = platformProductName;
+export const platformApiLevel = platformApi;
+export const osName = ExpoDevice ? ExpoDevice.osName : null;
+export const totalMemory = ExpoDevice ? ExpoDevice.totalMemory : null;
+export const isDevice = ExpoDevice ? ExpoDevice.isDevice : null;
+export const supportedCpuArchitectures = ExpoDevice ? ExpoDevice.supportedCpuArchitectures : null;
+export const osBuildId = ExpoDevice ? ExpoDevice.osBuildId : null;
+export const osVersion = ExpoDevice ? ExpoDevice.osVersion : null;
+export const deviceName = ExpoDevice ? ExpoDevice.deviceName : null;
+export const osInternalBuildId = ExpoDevice ? ExpoDevice.osInternalBuildId : null;
+
+export async function hasPlatformFeatureAsync(feature: string): Promise<boolean> {
+  if (!ExpoDevice.hasPlatformFeatureAsync) {
+    throw new UnavailabilityError('expo-device', 'hasPlatformFeatureAsync');
+  }
+  return await ExpoDevice.hasPlatformFeatureAsync(feature);
+}
+
+export async function getPlatformFeaturesAsync(): Promise<string[]> {
+  if (!ExpoDevice.getPlatformFeaturesAsync) {
+    throw new UnavailabilityError('expo-device', 'getPlatformFeaturesAsync');
+  }
+  return await ExpoDevice.getPlatformFeaturesAsync();
+}
+
+export async function getMaxMemoryAsync(): Promise<number> {
+  if (!ExpoDevice.getMaxMemoryAsync) {
+    throw new UnavailabilityError('expo-device', 'getMaxMemoryAsync');
+  }
+  return await ExpoDevice.getMaxMemoryAsync();
+}
+
+export async function isSideLoadingEnabled(): Promise<boolean> {
+  if (!ExpoDevice.isSideLoadingEnabled) {
+    throw new UnavailabilityError('expo-device', 'isSideLoadingEnabled');
+  }
+  return await ExpoDevice.isSideLoadingEnabled();
+}
+
+export async function getUptimeAsync(): Promise<number> {
+  if (!ExpoDevice.getUptimeAsync) {
+    throw new UnavailabilityError('expo-device', 'getUptimeAsync');
+  }
+  return await ExpoDevice.getUptimeAsync();
+}
+
+export async function isRootedExperimentalAsync(): Promise<boolean> {
+  if (!ExpoDevice.isRootedExperimentalAsync) {
+    throw new UnavailabilityError('expo-device', 'isRootedExperimentalAsync');
+  }
+  return await ExpoDevice.isRootedExperimentalAsync();
+}
+
+export async function getDeviceTypeAsync(): Promise<DeviceType> {
+  if (!ExpoDevice.getDeviceTypeAsync) {
+    throw new UnavailabilityError('expo-device', 'getDeviceTypeAsync');
+  }
+  const deviceType = await ExpoDevice.getDeviceTypeAsync();
+  switch(deviceType) {
+    case deviceType === DeviceType.PHONE: return DeviceType.PHONE;
+    case deviceType === DeviceType.TABLET: return DeviceType.TABLET;
+    case deviceType === DeviceType.TV: return DeviceType.TV;
+    case deviceType === DeviceType.DESKTOP: return DeviceType.DESKTOP;
+    default: return DeviceType.UNKNOWN;
```

Comment:
Is this `switch` necessary? Maybe you could use something like `if (!Object.values(DeviceType).includes(deviceType)) { return DeviceType.UNKNOWN; }`?

Reply:
Because we returned string from the native codes (iOS and Android), so we might want to check which enum type the returned string belongs to.